troposphere/__init__.py

```diff
@@ -509,11 +509,14 @@ def __init__(self, data: Any) -> None:
 
 class FindInMap(AWSHelperFn):
     def __init__(
-        self, mapname: object, toplevelkey: object, secondlevelkey: object
+        self, mapname: object, toplevelkey: object, secondlevelkey: object, defaultvalue: Optional[object] = None
     ) -> None:
-        self.data = {
-            "Fn::FindInMap": [self.getdata(mapname), toplevelkey, secondlevelkey]
-        }
+        find_in_map_details = [self.getdata(mapname), toplevelkey, secondlevelkey]
+
+        if defaultvalue is not None:
+            find_in_map_details.append(defaultvalue)
```

Comment:
Thank you for this change. If I'm reading the documentation correctly, shouldn't this be:
```
    find_in_map_details.append({'DefaultValue': defaultvalue})
```
And corresponding changes in the tests.

Reply:
My original thought was that the fourth item is essentially a dict type which would mean inputting "{'DefaultValue': defaultvalue}" in the template itself, but I suppose this is cleaner, I updated the PR.